app/controllers/api/notifications_controller.rb

```diff
@@ -0,0 +1,18 @@
+module Api
+  class NotificationsController < BaseController
+    def notifications_search_conditions
+      {:user_id => @auth_user_obj.id}
+    end
+
+    def find_notifications(id)
+      @auth_user_obj.notification_recipients.find(id)
+    end
+
+    def mark_as_seen_resource_notifications(type, id = nil, _data = nil)
```

Comment:
I have a test for that case called
```
it 'rejects on notification that is not owned by current user' do
```

We use `resource_search` here. The `resource_search` uses `find_notifications` that returns only those notifications that are owned by user.